server/controllers/users/getUsersByInterest.js

```diff
@@ -0,0 +1,32 @@
+const {
+  users: { getUsersByInterest: getUsers },
+  languages: { getLanguageById },
+} = require('../../database/queries');
+const { formatLanguages } = require('../../helpers/formatLanguages');
+
+exports.getUsersByInterest = (req, res, next) => {
+  let users;
+  const { id } = req.params;
+  // Validate the id
+  if (!Number(id)) return next({ code: 400 });
+  // Calling the query
+  getUsers(id)
+    .then((result) => {
+      users = result.rows;
+      return users.map((user) => getLanguageById(user.native_lang_id));
+    })
+    .then((promises) => Promise.all(promises))
+    .then((langs) => formatLanguages(users, langs, 'native_lang_id'))
+    .then((result) => {
+      users = result;
+      return users.map((user) => getLanguageById(user.learning_lang_id));
```

Comment:
You are doing three DB queries to get the results, which is three roundtrips to the database. It should be possible to combine it into a single query so that there is just a single DB roundtrip.
You would have to add a third inner join to the getUsersByInterest query between the User and Language tables. The join would have to be `WHERE lang.id = user.native_lang_id OR lang.id = user.learning_lang_id`. This should also allow you to get rid of the `formatLanguages` function.
It's a bit more complicated so up to you if you wanna spend time on that :)

Reply:
Yes it makes sense, I tried to do do it but I get repeated data in the table I get out of the query so I have to make another format function to get rid of the repeated data
- So I will refactor this code later because now we need this pr to be merged